### DISCUSSION TITLE
method `textDocument/inlayHint` is not supported by any of the servers registered for the current buffer

### DISCUSSION THREAD

**Discussion by nyngwang** [neovim/neovim#24403] [Q&A] [answered]
I use this new upstream API `get_clients` with `method` fileter to toggle inlayhint:

```lua
vim.api.nvim_create_autocmd({ 'InsertEnter' }, {
  group = curfile_augroup,
  callback = function ()
    if vim.bo.buftype ~= '' then return end
    if vim.bo.filetype:match('git.*') then return end
    if #vim.lsp.get_clients({ bufnr=0, 'textDocument/inlayHint' }) > 0 then
      vim.lsp.inlay_hint(0, true)
    end
  end,
})
vim.api.nvim_create_autocmd({ 'InsertLeave' }, {
  group = curfile_augroup,
  callback = function ()
    if vim.bo.buftype ~= '' then return end
    if vim.bo.filetype:match('git.*') then return end
    if #vim.lsp.get_clients({ bufnr=0, 'textDocument/inlayHint' }) > 0 then
      vim.defer_fn(function ()
        vim.lsp.inlay_hint(0, false)
      end, 300)
    end
  end,
})
```

and I got this error with `pyright`:

```
method textDocument/inlayHint is not supported by any of the servers registered for the current buffer
```

Does anyone know what I did wrong? I have added `if #vim.lsp.get_clients({ bufnr=0, 'textDocument/inlayHint' }) > 0 then` when the `vim.lsp.inlay_hink` still gets called?

**Comment by seandewar** [comment, marked as answer]
You're not setting the `method` key in the table.

E.g: `vim.lsp.get_clients({bufnr=0, method="textDocument/inlayHint"})`.

**Reply by nyngwang** [reply]
Is it just me or pyright is reluctant to give me some hint?

**Reply by seandewar** [reply]
I don't use pyright personally, but as hinted at from the original warning from `vim.lsp.inlay_hint()`, it seems that [it doesn't support them](https://github.com/microsoft/pyright/issues/4325) (also see [here](https://github.com/microsoft/pylance-release/issues/4169)).

Interestingly though, it seems coc.nvim gets around that restriction [with some trickery](https://www.reddit.com/r/neovim/comments/12ebnz7/how_to_get_inlay_hints_working_with_pyright/jfae4k2/).

**Reply by nyngwang** [reply]
@seandewar PyBad, PyWrong. and thank you!

**Comment by ksajan** [comment]
The same is happening with me When I try to go to a function implementation it says `textDocument/definition` and it is happening for all the languages that I have tried which are Lua, Typescript, python, and Golang.
